assets/src/scripts/store/network.spec.js

```diff
@@ -0,0 +1,114 @@
+        describe('setNetworkList', () => {
+            const NETWORK_LIST = [
+                {
+                    rel: 'collection',
+                    href: 'https://labs-dev.wma.chs.usgs.gov/api/observations/collections/monitoring-locations?f=json',
+                    type: 'application/json',
+                    title: 'NWIS Monitoring Locations'
+                },
+                {
+                    rel: 'collection',
+                    href: 'https://labs-dev.wma.chs.usgs.gov/api/observations/collections/AHS?f=json',
+                    type: 'application/json',
+                    title: 'Arkansas Hot Springs National Park Network'
+                },
+                {
+                    rel: 'collection',
+                    href: 'https://labs-dev.wma.chs.usgs.gov/api/observations/collections/RTN?f=json',
+                    type: 'application/json',
+                    title: 'Real-Time Groundwater Level Network'
+                },
+                {
+                    rel: 'collection',
+                    href: 'https://labs-dev.wma.chs.usgs.gov/api/observations/collections/CRN?f=json',
+                    type: 'application/json',
+                    title: 'Climate Response Network'
+                },
+                {
+                    rel: 'collection',
+                    href: 'https://labs-dev.wma.chs.usgs.gov/api/observations/collections/AGL?f=json',
+                    type: 'application/json',
+                    title: 'Active Groundwater Level Network'
+                }
+        ];
+
+            it('expect network data to be updated', () => {
+                store.dispatch(
+                    Actions.setNetworkList(NETWORK_LIST));
+                const networkData = store.getState().networkData;
+
+                expect(networkData.networkList).toEqual(NETWORK_LIST);
+            });
+        });
+
+        describe('retrieveNetworkData', () => {
+            it('Expects that fetching urls have the siteno', () => {
+                store.dispatch(Actions.retrieveNetworkListData('12345678'));
+
+                expect(jasmine.Ajax.requests.count()).toBe(1);
+                expect(jasmine.Ajax.requests.at(0).url).toContain('USGS-12345678');
+            });
+
+            it('Expects the store to be updated on successful fetches', (done) => {
+                const promise = store.dispatch(Actions.retrieveNetworkListData('12345678'));
+                jasmine.Ajax.requests.at(0).respondWith({
+                    status: 200,
+                    responseText: MOCK_OBSERVATION_ITEM
+                });
+
+                promise.then(() => {
+                    const networkData = store.getState().networkData;
+
+                    expect(networkData.networkList).toEqual(
+                        JSON.parse(MOCK_OBSERVATION_ITEM).links.filter(function (link) {
+                            link['href'] = buildNetworkURL(link['href']);
```

Comment:
Not sure what this is doing here. This line won't affect the return value.

Reply:
Oddly enough it works, but it's not a good coding practice because it's obscuring the purpose of filter.  Regardless, it has been removed since the buildNetworkURL is now in the component and out of the store.